### PR TITLE
Knowledge Base: Expand Scripting API Functions Info

### PR DESCRIPTION
**Pull request by Yakvi**
Task card: https://github.com/storm-devs/storm-engine/projects/4#card-58081815

New functions described for the scripting language reference guide:

* Entity processing
* Layer processing
* Trig functions
* Random number generation

Changes mostly affected article 0003, but I also introduced article 0004 (entities) and expanded a bit article 0002.

### CODE REVIEW

File: kb/0002-scripting-overview.md [hunk 1]
```diff
@@ -47,6 +47,12 @@ The game goes through the following phases:
 * Upon startup, each file is analyzed and compiled into a single translation unit (`COMPILER::Compile`).
 * During the game runtime, compiler will analyze each token and act upon it by introducing new tokens, updating existing, or removing old ones. This way the program will ebb and flow until it reaches its exit code (`COMPILER::BC_Execute`).
 
+### Entities and Layers
+
+The game logic operates on the concepts of "entities". These are the objects which have a shape the compiler expects. It can be tangible things like sea, weather, grass, characters, but also intangible things like battle interface or sea AI.
```
Comment (np-c0mp1ete) [inline kb/0002-scripting-overview.md:52]: I guess `shape` doesn't fit here. Maybe properties/attributes/interface?

File: kb/0003-scripting-functions.md [hunk 2]
```diff
@@ -35,267 +36,393 @@ Below are the functions which make part of the compiler API. Each function has i
...
-    usage: 
-        SetTimeScale(0.0);
+        object CreateClass(string entityType); 
+    example:
+        object obj = CreateClass("CustomType"); // "invalid entity name"
+        object obj = CreateClass("Sky");        // OK
     ```
 
-NOTE(yakvi): Below is work-in-progress
+* **CreateEntity**: Similar to `CreateClass`, except it allows select attributes to be used directly by the engine.
```
Comment (np-c0mp1ete) [inline kb/0003-scripting-functions.md:74]: `allows select` sounds wrong

**Review by np-c0mp1ete — COMMENTED** [review]
Thanks a lot for the PR! I'll let @espkk review and approve it.

**Review by espkk — APPROVED** [review]
Seems good to me. Thank you!

**Pull request merged**